interfaces.go

```diff
@@ -3,8 +3,8 @@ package merkleTree
 // StorageEngine specifies how to store and lookup merkle tree nodes
 // and roots.  You can use a DB like Dynamo or SQL to do this.
 type StorageEngine interface {
```

Comment:
I'm trying to visualize how this is going to work for someone looking up the nodes. If `Node` isn't passed to `StoreNode` how will a user know if it's an iNode or a leaf node when looking it up?